test/CAS/Tests/AuthenticationTest.php

```diff
@@ -127,19 +127,17 @@ protected function tearDown()
      * Test that the user is redirected to the CAS server
      *
      * @return void
+     *
+     * @expectedException CAS_GracefullTerminationException
      */
     public function testRedirect()
     {
+        ob_start();
         try {
-            ob_start();
             $this->object->forceAuthentication();
-            $this->assertTrue(
-                false, 'Should have thrown a CAS_GracefullTerminationException.'
-            );
-        } catch (CAS_GracefullTerminationException $e) {
+        } catch (Exception $e) {
             ob_end_clean();
```

Comment:
This line should be in a `finally` block for proper cleanup. So this whole test can be (I haven't tested this though):

```php
        ob_start();
        $this->expectException(CAS_GracefullTerminationException::class);
        try {
            $this->object->forceAuthentication();
        } finally {
            ob_end_clean();
        } 
```

Reply:
I agree. However PHP did not add the `finally` keyword until 5.5. Here is the changelog:

https://www.php.net/ChangeLog-5.php

As above, phpCAS currently supports as far back as 5.4. Is it okay to drop these older end-of-life PHPs?

Reply:
Wow didn't know that before. So we may need `ob_end_flush()` in both `try` and `catch` block?

Reply:
Makes sense. I've made this change in the latest revision.